### PR TITLE
Test multigrid coarsening of an appctx coefficient

### PR DESCRIPTION
**Pull request by pbrubeck**
Adds a test that a `Function` in the application context is injected onto every coarse level, so a monolithic multigrid cycle can precondition Stokes with a variable viscosity.

Relevant to the discussion in #5428.

Written with Claude Code (Claude Opus 5).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_019rf1udfpr3o22nZbyxAdrN

### CODE REVIEW

File: tests/firedrake/multigrid/test_stokes_gmg.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    x, y = SpatialCoordinate(mesh)
+    f = as_vector([sin(pi*x)*cos(pi*y), -cos(pi*x)*sin(pi*y)])
+    F = (inner(2*mu*sym(grad(u)), sym(grad(v)))
+         - inner(p, div(v))
+         - inner(div(u), q)
+         - inner(f, v))*dx
+
+    # The pressure is fixed by the natural boundary condition on side 4.
+    bcs = [DirichletBC(Z.sub(0), zero(), (1, 2, 3))]
```
Comment (pbrubeck) [inline tests/firedrake/multigrid/test_stokes_gmg.py]:
```suggestion
    bcs = [DirichletBC(Z.sub(0), 0, (1, 2, 3))]
```

**Review by connorjward — APPROVED** [review]
I have checked and this does indeed pass on `main` and fail on my branch.
